harvester/harvest/alma.py

```diff
@@ -89,28 +90,28 @@ def parse_marc_records_from_files(self, filepaths: list[str]) -> Iterator[MARCRe
                     while element.getprevious() is not None:
                         del element.getparent()[0]
 
-    def _get_latest_full_run_date(self, filepaths: list[str]) -> str:
+    def _get_latest_full_run_date(self, filepaths: list[str]) -> str | None:
         """Get the date from the latest Alma full extract."""
         filtered_filepaths = self._filter_filepaths_by_harvest_type(filepaths)
-        extracted_dates = sorted(
-            {
-                self._get_date_from_filepath(filepath)
-                for filepath in filtered_filepaths
-                if filepath
-            },
+        extracted_dates = set()
+        for filepath in filtered_filepaths:
+            if filepath_date := self._get_date_from_filepath(filepath):
+                extracted_dates.add(filepath_date)
+        sorted_extracted_dates = sorted(
+            extracted_dates,
             reverse=True,
         )
-        if not extracted_dates:
+        if not sorted_extracted_dates:
             raise AlmaCannotIdentifyLatestFullRunDateError
-        return extracted_dates[0]
+        return sorted_extracted_dates[0]
```

Comment:
Another variation, using `filter()` to remove `None` values, appears to satisfy the mypy linting:

```python
def _get_latest_full_run_date(self, filepaths: list[str]) -> str | None:
    """Get the date from the latest Alma full extract."""
    extracted_dates = sorted(
        filter(
            None,
            (
                self._get_date_from_filepath(filepath)
                for filepath in self._filter_filepaths_by_harvest_type(filepaths)
            ),
        ),
        reverse=True,
    )
    if not extracted_dates:
        raise AlmaCannotIdentifyLatestFullRunDateError
    return extracted_dates[0]
```

Disclaimer: I don't use `filter()` often, but it seems like potentially nice fit here?  But, also looked okay as-is!  Dealers choice, not blocking comment here.

Reply:
Good to know! I definitely think this is clear, concise, and keeps `mypy` happy. Change has been made.